lib/iex/lib/iex/evaluator.ex

```diff
@@ -256,6 +256,8 @@ defmodule IEx.Evaluator do
     {result, binding, env, scope} =
       :elixir.eval_forms(forms, state.binding, state.env, state.scope)
 
+    Process.delete(:iex_imported_paths)
```

Comment:
Unfortuntately, we can also invoke `import_file` from inside IEx, which means the pdict will never be cleaned. My suggesstion:

1. set the process dictionary before we import in the evaluator
2. in the import_file code, only do the check and change the pdict if the pdict was already set

This means we check unfortunately only the default import but it also means we don't leak the pdict.

Reply:
Oh, you always clean the pdict after any eval. Yeah, I don't like this, it feels we are leaking concerns. :'( It would be ok to do it exclusively in the import code (because that code is about importing ater all).